src/proxy/inbound.rs

```diff
@@ -227,16 +235,42 @@ impl Inbound {
         // At this point, we established the upstream connection and need to send a 200 back to the client.
         // we may still have failures at this point during the proxying, but we don't need to send these
         // at the HTTP layer.
+        // After sending a 200 back to the client, write the proxy protocol to the stream. This ensures
+        // that the server has all of the necessary information about the connection regardless of the protocol
+        // See https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt for more information about the
+        // proxy protocol.
         let send = req
             .send_response(build_response(StatusCode::OK))
             .and_then(|h2_stream| async {
+                // Required for a custom waypoint so that it doesn't have to support HBONE
                 if ri.inbound_protocol == AppProtocol::PROXY {
                     let Connection {
                         src, src_identity, ..
                     } = ri.rbac_ctx.conn;
-                    super::write_proxy_protocol(&mut stream, (src, ri.hbone_addr), src_identity)
-                        .instrument(trace_span!("proxy protocol"))
-                        .await?;
+                    let protocol_addr = match ri.hbone_addr {
+                        HboneAddress::SocketAddr(addr) => addr,
+                        HboneAddress::SvcHostname(_, _) => {
+                            // If the hbone_addr includes service hostname, we need to use the resolved svc IP address
+                            match ri.upstream_protocol_addr {
+                                Some(addr) => addr,
+                                None => {
+                                    // This should never happen, as we should have already validated the destination
+                                    // and resolved the hostname to an IP address.
+                                    return Err(Error::ConnectAddress(ri.hbone_addr.to_string()));
+                                }
+                            }
+                        }
+                    };
+                    // TODO(jaellio): Also include port?
```

Comment:
Yeah this should also include port

Reply:
Updated to include port. Fmt is <hostname>:<port>. Should this value instead be taken directly from the authority value  of the uri rather than manually formatted?